src/axom/multimat/multimat.hpp

```diff
@@ -216,6 +208,29 @@ class MultiMat
    */
   void setNumberOfCells(int num_cells);
 
+  /**
+   * \brief Sets the allocator ID to use for allocations.
+   *
+   * \param The allocator ID to use
+   */
+  void setAllocatorID(int alloc_id);
+
+  /**
+   * \brief Sets the allocator ID to use for Slam object allocations.
+   *  This should point to either a host Umpire pool if only being used on the
+   *  host, or a unified memory pool if also being used on the GPU.
```

Comment:
Does it have to use unified memory on the GPU, or can it use a device memory pool there?
If it's the former, we'll likely need to add support for the latter.

Reply:
I haven't tested it with a device-only allocator, but I suspect there's a lot more work to be done before that can be supported. At a minimum, the conversions between sparse/dense and cell-dominant/material-dominant need to be GPU loops.

Reply:
Thanks @publixsubfan.

Unfortunately, device-only allocators is necessary for the intended application, so I think we'll need to address that in a follow-up PR.